components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2926,6 +2926,12 @@ void RewardsServiceImpl::ProcessRewardsPageUrl(
       {});
 }
 
+void RewardsServiceImpl::RequestAdsEnabledPopupClosed(bool ads_enabled) {
```

Comment:
I'm confused, where does ads actually get enabled? Also why is this on RewardsService/RewardsServiceObserver instead of AdsService?

Reply:
this is a problem https://github.com/brave/brave-core/pull/9611/files#r716893487

Reply:
Added `EnableRewards()` call here if `ads_enabled` is true